environment.cpp

```diff
@@ -103,17 +103,19 @@ std::vector<double> create_n_inputs(std::uniform_real_distribution<double> dist,
 }
 
 
-void switch_env_function(environment &e)
+environment switch_env_function(environment e)
 {
     try
     {
         if(e.get_name_current_function()=='A'){
             e.change_env_function(e.get_env_function_B());
             e.switch_name_current_function();
+            return e;
```

Comment:
related to my last comment in the previous conversation. If you pass `environment` by ref here and simply run the memebr functions (you can or cannot return the environment afterwards) this free function will just modify environment.